### ISSUE TITLE
Distillation loss produces NaNs

### ISSUE THREAD

**Issue by khalidsaifullaah** [microsoft/LMOps#38]
For the following distillation loss that's being used in the repo, do you ever face NaN issues?
```
teacher_probs = F.softmax(teacher_logits, dim=-1, dtype=torch.float32)
inf_mask = torch.isinf(logits)
logprobs = F.log_softmax(logits, dim=-1, dtype=torch.float32)
prod_probs = torch.masked_fill(teacher_probs * logprobs, inf_mask, 0)
x = torch.sum(prod_probs, dim=-1).view(-1)
mask = (no_model_batch["label"] != -100).int()
distil_loss = -torch.sum(x * mask.view(-1), dim=0) / torch.sum(mask.view(-1), dim=0)
```

Although it works fine for the first couple of hundreds iteration but then starts producing NaNs

**Comment by t1101675** [comment]
What is the base model you use? Can you provide some information about the training log?
The NaN issue is probably caused by dividing by 0 in `-torch.sum(x * mask.view(-1), dim=0) / torch.sum(mask.view(-1), dim=0)` . You can check whether there exists any full-zero row in `mask`.

**Comment by khalidsaifullaah** [comment]
That was indeed the issue, looks like my tokenizer was truncating the inputs early. Thanks!
